### PR TITLE
[ALLUXIO-2393] Remove the Permission class.

### PR DESCRIPTION
**Pull request by jsimsa**
Addresses https://alluxio.atlassian.net/browse/ALLUXIO-2393

The motivation behind this PR is to decouple owner, group, and mode of path permissions as these attributes are often initialized through different means, which presents challenges for the `Permission` class API.

The main change in this PR is that the `Permission` class is removed. Its methods for initializing owner and group using login module and thrift client are moved to `SecurityUtils` and its methods for applying umask are moved to `Mode`.

The rest of the PR are mostly mechanical changes that replace uses of `Permission` with equivalent uses of owner, group, and mode.

### CODE REVIEW

**Comment by maobaolong** [comment]
#4704 My Pr blocked by the failed test too.

**Comment by jsimsa** [comment]
Jenkins, test this please

File: core/server/src/test/java/alluxio/master/file/PermissionCheckTest.java [hunk 1]
```diff
@@ -879,23 +893,29 @@ private String toExceptionMessage(String user, Mode.Bits bits, String path,
...
-            CreateFileOptions.defaults().setBlockSizeBytes(Constants.KB)
-                .setPermission(permissions.get(i)).setDefaultMode(false));
+            CreateFileOptions.defaults().setBlockSizeBytes(Constants.KB).setOwner(owner)
+                .setGroup(group).setMode(mode).setDefaultMode(false));
         inodes.add(inode);
       } else {
         Inode<?> inode = InodeDirectory.create(i + 1, i, (i + 1) + "",
-            CreateDirectoryOptions.defaults().setPermission(permissions.get(i)));
+            CreateDirectoryOptions.defaults().setOwner(owner).setGroup(group).setMode(mode));
+        inodes.add(inode);
```
Comment (jsimsa) [inline core/server/src/test/java/alluxio/master/file/PermissionCheckTest.java]: this was a bug

**Comment by jsimsa** [comment]
Jenkins, test this please

**Comment by ifcharming** [comment]
LGTM, thanks for the refactoring!

**Pull request merged**